src/Messaging/Events/TaskUpdateEvent.cs

```diff
@@ -50,7 +50,7 @@ public class TaskUpdateEvent : EventBase
         /// Gets or set the execution stats of the task.
         /// </summary>
         [JsonProperty(PropertyName = "taskStats")]
-        public Dictionary<string, object?>? ExecutionStats { get; set; }
+        public Dictionary<string, string> ExecutionStats { get; set; }
 
         /// <summary>
         /// Gets or set the failure reason of the task.
```

Comment:
could you look at cleaning up the warnings on this file 
`Non-nullable property 'ExecutionStats' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.`